hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -2394,16 +2396,22 @@ void transferBlock(ExtendedBlock block, DatanodeInfo[] xferTargets,
     
     int numTargets = xferTargets.length;
     if (numTargets > 0) {
-      final String xferTargetsString =
-          StringUtils.join(" ", Arrays.asList(xferTargets));
-      LOG.info("{} Starting thread to transfer {} to {}", bpReg, block,
-          xferTargetsString);
+      if (transferringBlock.contains(block)) {
+        LOG.warn(
```

Comment:
Can you explain why we are avoiding the block storm?
How big can this set get?

Reply:
Just like I said, when datanode's io is busy, DataTransfer will stuck for long time, the pendingReconstructBlock will timeout. Then a duplicated DataTransfer thread for same block will start, and the duplicated DataTransfer thread will make io busier. So I think we should avoid the duplicated DataTransfer.